packages/rum/src/domain/segmentCollection/segment.ts

```diff
@@ -12,31 +14,39 @@ export class Segment {
   private end: number
   private recordsCount: number
   private hasFullSnapshot: boolean
+  private viewStats: ViewReplayStats
 
   constructor(
     private worker: DeflateWorker,
     readonly context: SegmentContext,
     private creationReason: CreationReason,
     initialRecord: Record,
-    onWrote: (size: number) => void,
+    onWrote: (sizes: { compressed: number; raw: number }) => void,
     onFlushed: (data: Uint8Array, rawSize: number) => void
   ) {
     this.start = initialRecord.timestamp
     this.end = initialRecord.timestamp
     this.recordsCount = 1
     this.hasFullSnapshot = initialRecord.type === RecordType.FullSnapshot
 
+    this.viewStats = getOrCreateViewStats(context.view.id)
+    this.viewStats.segments_count += 1
+    this.viewStats.records_count += 1
+
+    let previousRawSize = 0
     const listener: DeflateWorkerListener = monitor(({ data }) => {
       if ('error' in data) {
         return
       }
 
       if (data.id === this.id) {
+        this.viewStats.segments_total_raw_size += data.rawSize - previousRawSize
```

Comment:
I am trying to understand why do we need this computation. is there a way to have the additional size directly in `data` instead of having to compute it each time? This would get rid of `previousRawSize` and reduce complexity I think.